src/lwc/geTemplateBuilder/geTemplateBuilder.js

```diff
@@ -333,7 +333,7 @@ export default class geTemplateBuilder extends NavigationMixin(LightningElement)
         if (modalData.action === SAVE) {
             let formSections = mutable(this.formSections);
             let formSection = formSections.find((fs) => { return fs.id === modalData.section.id });
-
+            formSection.defaultDisplayMode = modalData.section.defaultDisplayMode;
```

Comment:
@vmartinov-sforg This was the problem, I mistakenly reverted this line so the `defaultDisplayMode` was not getting saved to the Template JSON. It is fixed now.